File: app/src/main/java/com/example/grocer21/DatabaseViewModel.java
package com.example.grocer21;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.example.grocer21.Database.Allergy;
import com.example.grocer21.Database.AppRepository;
import com.example.grocer21.Database.Food;

import java.util.List;

public class DatabaseViewModel extends AndroidViewModel {

    private AppRepository appRepository;
    public final LiveData<List<Food>> allFoods;
    public final LiveData<List<Allergy>> allAllergies;

    public DatabaseViewModel(@NonNull Application application) {
        super(application);
        appRepository = new AppRepository(application);
        allFoods = appRepository.getAllFoods();
        allAllergies = appRepository.getAllAllergies();
    }

    LiveData<List<Food>> getAllFoods() {
        return allFoods;
    }

    public void insert(Food food) {
        appRepository.insert(food);
    }

    public LiveData<List<Allergy>> getAllAllergies() {
        return allAllergies;
    }
}

File: app/src/main/java/com/example/grocer21/DatabaseViewModel.kt
package com.example.grocer21

import android.app.Application
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.grocer21.database.AppDatabase
import com.example.grocer21.database.entities.Allergies
import com.example.grocer21.database.entities.Diets
import com.example.grocer21.database.entities.Foods
import kotlinx.coroutines.launch

internal class DatabaseViewModel(application: Application) : ViewModel() {
    private val repository: AppRepository
    val allFoods: LiveData<List<Foods>>
    val allAllergies: LiveData<List<Allergies>>
    val allDiets: LiveData<List<Diets>>

    init {
    }

    init {
        val foodsDao = AppDatabase.getDatabase(application).foodsDao()
        val allergiesDao = AppDatabase.getDatabase(application).allergiesDao()
        val dietFoodsDao = AppDatabase.getDatabase(application).dietFoodsDao()
        val dietsDao = AppDatabase.getDatabase(application).dietsDao()
        val foodAllergensDao = AppDatabase.getDatabase(application).foodAllergensDao()
        val profileDao = AppDatabase.getDatabase(application).profileDao()
        val shoppingListDao = AppDatabase.getDatabase(application).shoppingListDao()
        val userAllergiesDao = AppDatabase.getDatabase(application).userAllergiesDao()
        val userDietsDao = AppDatabase.getDatabase(application).userDietsDao()
        repository = AppRepository(
                allergiesDao,
                dietsDao,
                foodsDao)

        allFoods = repository.allFoods
        allAllergies = repository.allAllergies
        allDiets = repository.allDiets

    }

    fun insertWrapper(food: Foods) {
        viewModelScope.launch {
            insert(food)
        }
    }

    private fun insert(food: Foods) {
        repository.insert(food)
    }

}
